src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3207,6 +3214,56 @@ private static bool HasNonConstantInitializer(ArrayBuilder<ImmutableArray<FieldO
             AddInitializers(builder.StaticInitializers, staticInitializers);
         }
 
+        private void CollectFieldsFromGlobalDeconstruction(ArrayBuilder<Symbol> builder, VariableComponentSyntax variableComponent,
+            SourceMemberContainerTypeSymbol containingType, VariableComponentAssignmentSyntax assignment)
+        {
+            switch (variableComponent.Kind())
+            {
+                case SyntaxKind.TypedVariableComponent:
+                    CollectFieldsFromGlobalDeconstruction(
+                        builder,
+                        ((TypedVariableComponentSyntax)variableComponent).Designation,
+                        containingType,
+                        assignment);
+                    break;
+
+                case SyntaxKind.ParenthesizedVariableComponent:
+                    foreach (VariableComponentSyntax variable in ((ParenthesizedVariableComponentSyntax)variableComponent).Variables)
+                    {
+                        CollectFieldsFromGlobalDeconstruction(builder, variable, containingType, assignment);
+                    }
+                    break;
+
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(variableComponent.Kind());
+            }
+        }
+
+        private void CollectFieldsFromGlobalDeconstruction(ArrayBuilder<Symbol> builder, VariableDesignationSyntax designation,
+            SourceMemberContainerTypeSymbol containingType, VariableComponentAssignmentSyntax assignment)
+        {
+            switch (designation.Kind())
+            {
+                case SyntaxKind.SingleVariableDesignation:
+                    var field = SourceMemberFieldSymbolFromDesignation.Create(containingType, (SingleVariableDesignationSyntax)designation,
+                        DeclarationModifiers.Private, null, assignment);
+
+                    builder.Add(field);
+
+                    break;
+
+                case SyntaxKind.ParenthesizedVariableDesignation:
+                    foreach (VariableDesignationSyntax variable in ((ParenthesizedVariableDesignationSyntax)designation).Variables)
+                    {
+                        CollectFieldsFromGlobalDeconstruction(builder, variable, containingType, assignment);
+                    }
+                    break;
```

Comment:
Consider extracting a `CollectFieldsFromGlobalDeconstruction(..., ParenthesizedVariableDesignationSyntax, ...)` for use here and method above.